src/utils/generateEventLink.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    CalendarEvent as CalendarLinkEvent,
+    google,
+    outlook,
+    ics,
+} from 'calendar-link';
+import { parse } from 'date-fns';
+
+export const enum CalendarType {
```

Comment:
I saw this practice but honestly, I did not like that approach, I would prefer to keep interfaces/types where there are used and export if they are reusable, bcz one-day `types.d.ts` will become huge so it brings unreadability and hard to maintain

Reply:
+ I suppose CalendarType and CalendarEvent will fo to another file, where this object will start fulfilling

Reply:
definitely, it changes a situation if there are many interfaces in the file, so we can create a type file near a source file and export from there, but there is not a lot of interfaces so far

Reply:
Ok, it's not critical. So leave it as it is now.

Regarding `types.d.ts` becoming huge, you may just split it into meaningful structure, and/or bring namespaces, so it's not a problem.

> plus, I suppose CalendarType and CalendarEvent will go to another file, where this object will start fulfilling

And with `.d.ts` files you wouldn't care about importing/exporting, it would be available everywhere ;)

Reply:
> Regarding types.d.ts becoming huge, you may just split it into meaningful structure, and/or bring namespaces, so it's not a problem.

huge file stays as a huge file 👍 

And with .d.ts files you wouldn't care about importing/exporting, it would be available everywhere ;)

> yes, but I would prefer to import/export to avoid overwriting reserved intefaces

Reply:
Sure, I understand your points, keep it as you prefer.

> huge file stays as a huge file 👍

you may have many `.d.ts` files nicely structured :)